### PR TITLE
File upload

### PR DESCRIPTION
**Pull request by RokasKil**
It works I think, reik pririšt prie user bet jo dar nėra so it is what it is.

### CODE REVIEW

File: backend/src/main/java/lt/vu/tube/entity/Video.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    private VideoStatusEnum status = VideoStatusEnum.NONE;
+
+    private String mime;
+
+    // TODO: add these whenever we get users
+    //@ManyToOne
+    //private User owner;
+
+
+    public UUID getId() {
```
Comment (agosu) [inline backend/src/main/java/lt/vu/tube/entity/Video.java:40]: We are not using Lombok? Not saying that we should, just that it was a requirement for first personal task.

File: backend/src/main/java/lt/vu/tube/response/VideoUploadResponse.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        this.id = id;
+    }
+
+    public static VideoUploadResponse fail (String message) {
+        return new VideoUploadResponse("fail", message);
+    }
+
+    public static VideoUploadResponse success (String message, UUID id) {
+        return new VideoUploadResponse("success", message, id);
+
```
Comment (agosu) [inline backend/src/main/java/lt/vu/tube/response/VideoUploadResponse.java]: empty line

File: backend/src/main/java/lt/vu/tube/util/AWSS3Utils.java [hunk 3]
```diff
@@ -155,4 +166,34 @@ public void setDeleteObjectResponse(DeleteObjectResponse deleteObjectResponse) {
         }
     }
 
+    private static class MultipartUploadPartContainer {
```
Comment (agosu) [inline backend/src/main/java/lt/vu/tube/util/AWSS3Utils.java:169]: We have several classes in one file?

File: backend/src/main/java/lt/vu/tube/web/VideoController.java [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    @Autowired
+    AWSCloudFrontUtils cloudFrontUtils;
+
+    @Autowired
+    EntityManager entityManager;
+
+    @Autowired
+    VideoDao videoDao;
+
+    Logger logger = Logger.getLogger(getClass().toString());
```
Comment (agosu) [inline backend/src/main/java/lt/vu/tube/web/VideoController.java]: Usually loggers are private static final

**Review by agosu — APPROVED** [review]

File: backend/src/main/java/lt/vu/tube/dao/VideoDao.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package lt.vu.tube.dao;
+
+import lt.vu.tube.entity.Video;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+import java.util.UUID;
+
+@Repository
+public interface VideoDao extends CrudRepository<Video, UUID> {
```
Comment (Bernardas-Labutis) [inline backend/src/main/java/lt/vu/tube/dao/VideoDao.java]: Probably should name it VideoRepository

File: backend/src/main/java/lt/vu/tube/entity/Video.java (hunk 1 above)
Comment (Bernardas-Labutis) [reply backend/src/main/java/lt/vu/tube/entity/Video.java:40]: When I tried using lombok in a jpa entity, it broke it, maybe we should avoid using lombok here 🗡️

File: backend/src/main/java/lt/vu/tube/util/AWSS3Utils.java (hunk 3 above)
Comment (Bernardas-Labutis) [reply backend/src/main/java/lt/vu/tube/util/AWSS3Utils.java:169]: As it is a class only used in this class, an inner class is suitable for this

**Review by Bernardas-Labutis — APPROVED** [review]

File: backend/src/main/java/lt/vu/tube/entity/Video.java (hunk 1 above)
Comment (agosu) [reply backend/src/main/java/lt/vu/tube/entity/Video.java:40]: Totally agree

File: backend/src/main/java/lt/vu/tube/dao/VideoDao.java (hunk 5 above)
Comment (RokasKil) [reply backend/src/main/java/lt/vu/tube/dao/VideoDao.java]: Fixed

File: backend/src/main/java/lt/vu/tube/web/VideoController.java (hunk 4 above)
Comment (RokasKil) [reply backend/src/main/java/lt/vu/tube/web/VideoController.java]: Fixed

File: backend/src/main/java/lt/vu/tube/response/VideoUploadResponse.java (hunk 2 above)
Comment (RokasKil) [reply backend/src/main/java/lt/vu/tube/response/VideoUploadResponse.java]: Fixed

**Pull request merged**